### PR TITLE
Update pytorch (1.12.1), torchaudio (1.13.1), torchvision (0.12.1)

### PR DESCRIPTION
**Pull request by mweinelt**
https://github.com/pytorch/pytorch/releases/tag/v1.12.0
https://github.com/pytorch/pytorch/releases/tag/v1.12.1

https://github.com/pytorch/audio/releases/tag/v0.12.0
https://github.com/pytorch/audio/releases/tag/v0.12.1

https://github.com/pytorch/vision/releases/tag/v0.13.0
https://github.com/pytorch/vision/releases/tag/v0.13.1

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by junjihashimoto — APPROVED** [review]
LGTM!
Great job!

File: pkgs/development/python-modules/torch/default.nix [hunk 1]
```diff
@@ -243,15 +238,17 @@ in buildPythonPackage rec {
...
   doCheck = false;
+
   pythonImportsCheck = [
     "torch"
   ];
 
+  checkInputs = [ hypothesis ninja psutil ];
+
   checkPhase = with lib.versions; with lib.strings; concatStringsSep " " [
+    "runHook preCheck"
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/torch/default.nix:251]: Can we make this a multi line string?

**Review by SuperSandro2000 — APPROVED** [review]

**Review by tpwrules — APPROVED** [review]
Passed building and a very basic smoke test on `aarch64-darwin` and `x86_64-linux` with CUDA acceleration.

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -10918,14 +10918,14 @@ in {
   toposort = callPackage ../development/python-modules/toposort { };
 
   torch = callPackage ../development/python-modules/torch {
+    cudaPackages = pkgs.cudaPackages_11_6;
```
Comment (mweinelt) [inline pkgs/top-level/python-packages.nix:10921]: @tpwrules I think this is what you asked for?

Comment (mweinelt) [reply pkgs/top-level/python-packages.nix]: Dropped per discussion in the nix cuda room.

**Pull request merged**